### PR TITLE
Feature: Design and code improvements

### PR DESCRIPTION
**Pull request by dimified**
In this branch you will find the first improvements that generally address issues and styling fixes for the current state of the application repliear. Before starting with the challenge I have seen a lot of issues that are fixed in this branch. Please see further screenshots for more details. I have focussed on maintaining functionality while improving specific details. This branch might also be a candidate for a Pull Request in the main repository.

### Fixes

- ✅ Fixes initialisation issue with rows[0] being undefined
- ✅ Fixes issue with selecting multiple filters
- ✅ Updates to the newest modules
- ✅ Defines new project setup with environment variables

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -19,19 +19,31 @@ Get the Pusher environment variables from the ["App Keys" section](https://i.img
 
 **Note:** These instructions assume you installed PostgreSQL via Postgres.app on MacOS. If you installed some other way, or configured PostgreSQL specially, you may additionally need to set the `PGUSER` and `PGPASSWORD` environment variables.</p>
 
+1. Create a local env file
+
+Create `.env.local` in the root of your project and copy & paste following environment variables that will be publicly available on runtime.
```
Comment (dimified) [inline README.md:24]: New approach to run the application with env files

File: backend/data.ts [hunk 2]
```diff
@@ -43,7 +43,8 @@ async function getSchemaVersion(executor: Executor) {
   const qr = await executor(
     `select value from meta where key = 'schemaVersion'`
   );
-  return qr.rows[0].value;
+
+  return qr.rows[0]?.value;
```
Comment (dimified) [inline backend/data.ts:47]: Fixing issue on startup

File: frontend/app.tsx [hunk 3]
```diff
@@ -65,6 +65,9 @@ class Filters {
     this.hasNonViewFilters = false;
     if (statusFilter) {
       this._issuesStatuses = new Set<Status>();
+
+      if (typeof statusFilter !== "string") return;
```
Comment (dimified) [inline frontend/app.tsx:69]: Fixing issue on startup

File: frontend/app.tsx [hunk 4]
```diff
@@ -82,6 +85,9 @@ class Filters {
 
     if (priorityFilter) {
       this._issuesPriorities = new Set<Priority>();
+
+      if (typeof priorityFilter !== "string") return;
```
Comment (dimified) [inline frontend/app.tsx:89]: When data was already set, it is an array and not a string anymore.

File: frontend/button.tsx [hunk 5]
```diff
@@ -0,0 +1,16 @@
+import type { ReactComponent } from "react-hotkeys";
+
+interface Props {
+  onClick: () => unknown;
+  icon: ReactComponent;
+  label: string;
+}
+
+export const Button: React.FC<Props> = ({ onClick, label, icon: Icon }) => (
+  <button
```
Comment (dimified) [inline frontend/button.tsx:10]: Created own button component before using Radix or other library

File: frontend/filter-menu.tsx [hunk 6]
```diff
@@ -101,7 +90,7 @@ const FilterMenu = ({ onSelectStatus, onSelectPriority }: Props) => {
       <button
         className="px-1 py-0.5 ml-3 border border-gray-600 border-dashed rounded text-white hover:text-gray-50 focus:outline-none"
         ref={setFilterRef}
-        onMouseDown={handleDropdownClick}
+        onClick={handleDropdownClick}
```
Comment (dimified) [inline frontend/filter-menu.tsx:93]: Better usability as context (right-click) menu is otherwise not possible.

File: frontend/input.tsx [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+
+interface Props {
+  onChange: ChangeEventHandler | undefined;
+  onFocus?: VoidFunction;
+  onBlur?: VoidFunction;
+  icon: ReactComponent;
+  placeholder: string;
+}
+
+export const Input: React.FC<Props> = ({
```
Comment (dimified) [inline frontend/input.tsx:12]: Own input component to have styling fixed and simplify things.

File: frontend/left-menu.tsx [hunk 8]
```diff
@@ -83,63 +84,30 @@ const LeftMenu = ({
...
-              onMouseDown={async () => {
-                await setLayoutViewParams({ view: "board", iss: null });
-                onCloseMenu && onCloseMenu();
-              }}
-            >
-              <span>Board</span>
-            </div>
-          </ItemGroup>
+          <ItemGroup
+            title="React Issues"
```
Comment (dimified) [inline frontend/left-menu.tsx:101]: Moved everything into a function component

**Comment by aboodman** [comment]
I'm confused as to why this PR was closed?

**Pull request closed without merging**